sonic-chassisd/scripts/chassisd

```diff
@@ -158,7 +161,20 @@ class ModuleConfigUpdater(logger.Logger):
         if (admin_state == MODULE_ADMIN_DOWN) or (admin_state == MODULE_ADMIN_UP):
             # Setting the module to administratively up/down state
             self.log_info("Changing module {} to admin {} state".format(key, 'DOWN' if admin_state == MODULE_ADMIN_DOWN else 'UP'))
-            try_get(self.chassis.get_module(module_index).set_admin_state, admin_state, default=False)
+            # Acquire the lock before submitting the callback function
+            with self.lock:
```

Comment:
Can you please explain the logic here? Why the thread is needed? 
The lock is taken in the parent thread and is unlocked immediately after the thread is started. It is not available for the child threads and can't protect data.

Reply:
This is not required. Reverted the change.

Reply:
This is still not resolved...

Reply:
It works without that.  However, the config change notification is sent sequentially to each DPU and takes severalX time depending on the number of DPUs.  As we use this mainly for time consuming startup/shutdown it is very slow.  Having this concurrent triggering solves the issue.  If you see any issue we can fix it.  Let me know if there is a better way to solve this.  I'm open to your suggestion.

Reply:
@rameshraghupathy this comment is still not resolved. Please do not mark it as resolved.

The lock is taken before the thread is stated. The thread doesn't take the lock at all.  Why do we need to use the lock here at all? What resource are you trying to protect?